src/webui/static/app/epg.js

```diff
@@ -1166,13 +1166,11 @@ tvheadend.epg = function() {
     epgFilterTitle.on('valid', function(c) {
         var value = c.getValue();
 
-        if (value.length < 1)
-            value = null;
-
-        if (epgStore.baseParams.title !== value) {
+        if (value.length < 1) {
+            delete epgStore.baseParams.title;
+        else if (epgStore.baseParams.title !== value)
             epgStore.baseParams.title = value;
-            epgView.reset();
-        }
+        epgView.reset();
```

Comment:
It would be probably better to call reset() when title changes, thus do 'delete title' in the second condition.

Reply:
Thanks for your comment, perexg. Then I need to modify the patch to call reset() in both 1st and 2nd condition. Calling reset() only in the 2nd condition does not refresh the list when user type some search string and then delete it to be empty.